lib/server.js

```diff
@@ -32,6 +32,10 @@ var Router = util.safeRequire('react-router');
 var match = Router.match;
 var RoutingContext = Router.RoutingContext;
 
+//for creating a URL
+var createMemoryHistory = require('history').createMemoryHistory;
+var createLocation = createMemoryHistory().createLocation;
```

Comment:
Sorry for the bad explanation

The above declaration is not necessary, it can easily substituted by a single line. I will refactor accordingly.
```js
var createLocation = require('history/lib/createLocation');
``` 
as mentioned in the [ServerRendering.md](https://github.com/rackt/react-router/blob/master/docs/guides/advanced/ServerRendering.md) from react-router
```js
match({ routes:createOptions.routes, location:createLocation(thing)},
          function(error, redirectLocation, renderProps) {
```